### ISSUE TITLE
PHP SDK codegen exception for 'default' response 

### ISSUE THREAD

**Issue by adhocore** [swagger-api/swagger-codegen#5586]
##### Description
[PHP] Bug generating `*Api::*WithHttpInfo()`

For `default` response in the [schema](https://pastebin.com/fcwebCHe), `TestApi::testWithHttpInfo()` is generated as:
```php
try {
  list($response, $statusCode, $httpHeader) = $this->apiClient->callApi(.......);
  return [..., $statusCode, $httpHeader];
} catch (ApiException $e) {
  switch ($e->getCode()) {
    case 200:
      $data = ...; break;
    case 0:
      $data = ...; break;
  }
  throw $e;
}
```
Should it not be like this one:
```php
switch ($e->getCode()) {
  case 200:
    $data = ...; break;
  default:
    $data = ...; break;
}
```

##### Swagger-codegen version

2.2.2

##### Swagger declaration file content or url

https://pastebin.com/fcwebCHe

##### Command line used for generation

java -jar swagger-codegen-cli.jar generate -i resources/schema/spec.json -l php -Dmodel -o sdk/php

##### Suggest a Fix

Thanks :)

**Comment by wing328** [comment]
@adhocore I believe non-2xx response will result in ApiException.

**Comment by adhocore** [comment]
i mean it is not about `2xx` please see the generated code section:
```php
case 200:
      $data = ...; break;
    case 0:
      $data = ...; break;
```
there is very slim chance that non 2xx code is but `case 0`. 0 is not even http status code. since we are defaulting anything except 200 to be caught as error, should it not be
```php
switch ($e->getCode()) {
  case 200:
    $data = ...; break;
  default:
    $data = ...; break;
}
```

**Comment by wing328** [comment]
The `code` usually refers to HTTP status code but it can contain 0 if my memory serves me well for the situation in which the error happens in the client side (e.g. network issue that the client cannot reach the server)

**Comment by adhocore** [comment]
that is network disconnected issue but is this the only case for defaulting? ofcourse no. all errors be it 0 or 4xx or 5xx are caught by default: section and it resonates to the schema too. but 0? :)

**Comment by wing328** [comment]
@adhocore I think I know what you mean now. 'default` response was incorrectly shown as 0. Like what you said, it should be mapped to the `default` switch case.

Would you have time to contribute the fix? If yes, I can show you some good starting points.

**Comment by arnested** [comment]
I just had a look at this and it seems @wing328 fixed this earlier today in commit ef35b80bd873281266389941382a6ecea65eb17a.

**Comment by wing328** [comment]
@adhocore let us know if you still see issues in the latest master.
